### PR TITLE
ppc64le: Adding test for appendProtectionDevice

### PR DESCRIPTION
**Pull request by Amulyam24**
Fixes: #2038

Signed-off-by: Amulya Meka <amulmek1@in.ibm.com>

### CODE REVIEW

**Review by jodh-intel — APPROVED** [review]
Thanks @Amulyam24! We :heart: new tests :smile: 

lgtm

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @Amulyam24

File: src/runtime/virtcontainers/qemu_ppc64le_test.go [hunk 1]
```diff
@@ -50,3 +50,53 @@ func TestQemuPPC64leMemoryTopology(t *testing.T) {
...
+	ppc64le.(*qemuPPC64le).protection = sevProtection
+	devices, bios, err = ppc64le.appendProtectionDevice(devices, firmware)
+	assert.Error(err)
+	assert.Empty(bios)
+
+	//TDX protection
+	ppc64le.(*qemuPPC64le).protection = tdxProtection
+	devices, bios, err = ppc64le.appendProtectionDevice(devices, firmware)
+	assert.Error(err)
+	assert.Empty(bios)
```
Comment (fidencio) [inline src/runtime/virtcontainers/qemu_ppc64le_test.go:83]: Just to be sure I understand this.
We are only add such cases as a negative test, right?

**Review by fidencio — APPROVED** [review]
lgtm, thanks @Amulyam24!

File: src/runtime/virtcontainers/qemu_ppc64le_test.go (hunk 1 above)
Comment (Amulyam24) [reply src/runtime/virtcontainers/qemu_ppc64le_test.go:83]: yes @fidencio

**Pull request merged**